### PR TITLE
numbered lists RSC-665

### PR DESCRIPTION
**Pull request by glenhunter**
https://issues.sonatype.org/browse/RSC-665

Created `<ol>` example with Jason (who will update the Sketch files accordingly).

### CODE REVIEW

File: gallery/src/styles/NxList/NxListExamples.tsx [hunk 1]
```diff
@@ -59,6 +60,14 @@ const NxListsExamples = () =>
       default <NxCode>&lt;ul&gt;</NxCode> styling. These can be nested.
     </GalleryExampleTile>
 
+    <GalleryExampleTile title="NX Numbered List Example"
+                        id="nx-list-numbered-example"
+                        htmlExample={NxListNumberedCode}
+                        codeExamples={NxListNumberedCode}>
+      An <NxCode>nx-list</NxCode> demonstrating ordered (numbered) list styles more typical of
+      default <NxCode>&lt;ol&gt;</NxCode> styling. These can be nested.
```
Comment (rpokorny) [inline gallery/src/styles/NxList/NxListExamples.tsx:68]: Is there a limit to how far they can be nested?

File: lib/src/base-styles/_nx-lists.scss [hunk 2]
```diff
@@ -301,3 +301,61 @@
   margin-left: var(--nx-spacing-6x);
   max-width: var(--nx-width-text-wrapping);
 }
+
+.nx-list--numbered {
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-lists.scss:305]: Surely a lot of this duplicated the bulleted list styling? I'd want to see the code duplication minimized, probably by just using a comma selector (`.nx-list--numbered, .nx-list--bulleted`)

**Comment by rpokorny** [comment]
The indentation is dramatically different between bulleted and numbered lists.  Is that deliberate?

File: gallery/src/styles/NxList/NxListExamples.tsx (hunk 1 above)
Comment (glenhunter) [reply gallery/src/styles/NxList/NxListExamples.tsx:68]: > Is there a limit to how far they can be nested?

Like bulleted lists we aren't putting a limit on nesting, but we have only explicitly styled the first 3 levels

**Comment by glenhunter** [comment]
> The indentation is dramatically different between bulleted and numbered lists. Is that deliberate?

I brought this up with @JenniKobylski and Jason and they are fine with the differences

File: lib/src/base-styles/_nx-lists.scss (hunk 2 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-lists.scss:305]: > Surely a lot of this duplicated the bulleted list styling? I'd want to see the code duplication minimized, probably by just using a comma selector (`.nx-list--numbered, .nx-list--bulleted`)

Sure, consolidated here: https://github.com/sonatype/sonatype-react-shared-components/pull/616/commits/28945332181e9d2008eaf4f22fdd67c046d4ff66

File: gallery/src/styles/NxList/NxListExamples.tsx (hunk 1 above)
Comment (rpokorny) [reply gallery/src/styles/NxList/NxListExamples.tsx:68]: > Like bulleted lists we aren't putting a limit on nesting, but we have only explicitly styled the first 3 levels

Should we at least have some awareness of what it does if you go deeper?

**Comment by rpokorny** [comment]
`nx-list--numbered` needs docs

File: gallery/src/styles/NxList/NxListExamples.tsx (hunk 1 above)
Comment (glenhunter) [reply gallery/src/styles/NxList/NxListExamples.tsx:68]: > > Like bulleted lists we aren't putting a limit on nesting, but we have only explicitly styled the first 3 levels
> 
> Should we at least have some awareness of what it does if you go deeper?

Like the bulleted list it just uses the last defined `list-type` for all nested list items past the third. So in this case that's Roman numerals (I also verified this in the code).

**Comment by glenhunter** [comment]
> `nx-list--numbered` needs docs

Right: https://github.com/sonatype/sonatype-react-shared-components/pull/616/commits/fdbf6416493fbd43d803a9023efadc268d76d228

**Comment by rpokorny** [comment]
Needs visual test

File: lib/src/base-styles/_nx-lists.scss [hunk 3]
```diff
@@ -301,3 +301,38 @@
...
+    .nx-list {
+      padding-left: var(--nx-spacing-12x);
+    }
+
+    .nx-list__item {
+      list-style: lower-alpha outside;
+      padding-top: var(--nx-spacing-2x);
+
+      &::marker {
+        font-weight: bold;
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-lists.scss:325]:
```suggestion
        font-weight: 600;
```

**Comment by glenhunter** [comment]
> Needs visual test

Yes it does: https://github.com/sonatype/sonatype-react-shared-components/pull/616/commits/52fd309c7ae7656c9a3079f6a592c7c2b463ecf9

File: lib/src/base-styles/_nx-lists.scss [hunk 4]
```diff
@@ -301,3 +301,38 @@
   margin-left: var(--nx-spacing-6x);
   max-width: var(--nx-width-text-wrapping);
 }
+
+.nx-list--numbered {
+  padding: 0 0 0 var(--nx-spacing-12x);
+
+  .nx-list__item {
+    list-style: decimal outside;
+    padding: var(--nx-spacing-2x) 0 0 0;
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-lists.scss:310]: Seems to be redundant

File: lib/src/base-styles/_nx-lists.scss [hunk 5]
```diff
@@ -301,3 +301,38 @@
...
+      margin-left: 0;
+    }
+
+    .nx-list {
+      padding-left: var(--nx-spacing-12x);
+    }
+
+    .nx-list__item {
+      list-style: lower-alpha outside;
+      padding-top: var(--nx-spacing-2x);
```
Comment (rpokorny) [inline lib/src/base-styles/_nx-lists.scss:322]: Also seems to be redundant

**Review by rpokorny — APPROVED** [review]
+1 aside from two redundant styles

File: lib/src/base-styles/_nx-lists.scss (hunk 4 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-lists.scss]: > Seems to be redundant

https://github.com/sonatype/sonatype-react-shared-components/pull/616/commits/b907196c74622e842364d350d20bb69254016b12

File: lib/src/base-styles/_nx-lists.scss (hunk 5 above)
Comment (glenhunter) [reply lib/src/base-styles/_nx-lists.scss]: > Also seems to be redundant

https://github.com/sonatype/sonatype-react-shared-components/pull/616/commits/f7098ffe2db3c2932434de0284879526230ccacf

**Pull request merged**